### PR TITLE
fix(observability): ensure internal rate limiting is logged

### LINKED ISSUE #841: Log rate limit errors, internal and external
If a sink is being rate-limited either internally or externally we need to log what is happened to communicate to the user that data is being rate-limited.

Logging for internal rate limits is especially important since users can generally adjust setting to eliminate this.

### PR DESCRIPTION
**Pull request by lukesteensen**
Closes #841 

Tested locally and works as expected.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -169,6 +169,7 @@ fn main() {
             format!("vector={}", level),
             format!("codec={}", level),
             format!("file_source={}", level),
+            format!("tower_limit=trace"),
```
Comment (LucioFranco) [inline src/main.rs]: actually, do we want to use `{}` here?

Comment (lukesteensen) [reply src/main.rs]: There's only one trace event in the whole crate and it's at `trace` level, so it's the only level that will ever show anything.

I'd actually like for this to be at `warn`, but that seemed aggressive to hardcode into a library. If there were a way to translate into our desired levels, then I'd definitely use `{}` here.

Comment (LucioFranco) [reply src/main.rs]: Well this might happen more often this will enable it for every log level. That seems a bit wrong?

Comment (lukesteensen) [reply src/main.rs]: The alternative would be that it's only ever logged when we're at `trace` everywhere, which I don't think fixes the issue of these being visible during normal operation.

Comment (LucioFranco) [reply src/main.rs]: fair point, not sure how to solve that then :(

Comment (lukesteensen) [reply src/main.rs]: Have people thought much about appropriate log levels for libraries? We could change tower-limit to `warn` here, but I'm not sure that's something everyone would want.

Comment (LucioFranco) [reply src/main.rs]: @hawkw any thoughts?

**Comment by lukesteensen** [comment]
I'm going to merge for now so we get the functionality, but we can continue the discussion and revisit if we come up with something better.

**Pull request merged**